src/domUpdates.js

```diff
@@ -109,23 +105,64 @@ const domUpdates = {
     $('#money-spent').prepend(`You have spent a total of $${totalBookingCosts} on bookings.`);
 
     $('#date-picker').change(function(){
-      const date = $('#date-picker').val().replace(/-/g, '/');
+      const chosenDate = $('#date-picker').val().replace(/-/g, '/');
+      domUpdates.createUserRoomCards(roomData);
       domUpdates.filterAvailableRoomsByDate(chosenDate);
     });
 
     $('#manager-portal').hide();
   },
 
-
   hideLoginPage: () => {
     $('#login-page').hide();
   },
 
+  createUserRoomCards: (roomData) => {
+    let rooms = roomData;
+    $('#rooms-section').prepend(
+      `${rooms.map(room =>
+        `<div class="room-card" data-room-type=${room.roomType} data-room-number=${room.number}>
+          <h3>${room.roomType}</h3>
+          <div class="room-info">
+          <p>Room Number ${room.number}</p>
+          <p>Number of Beds: ${room.numBeds}</p>
+          <p>Bed Size: ${room.bedSize}</p>
+          <p>Bidet: ${room.hasBidet}</p>
+          <p>Cost Per Night: ${room.costPerNight}</p>
+        </div>
+        <button type="button" name="button">BOOK ROOM</button></button>
+      </div>`
+      ).join("")}`
+    );
+  },
+
   filterAvailableRoomsByDate: (chosenDate) => {
-    let avaiableRooms = hotel.findAviableRooms(chosenDate);
+    let roomCards = Array.from(document.querySelectorAll('.room-card'));
+    let availableRoomNumbers = hotel.findAvailableRooms(chosenDate).map(room => room.number);
 
+    const toHide = roomCards.filter(card => {
+      const cardNumber = parseInt(card.dataset.roomNumber);
+      return !availableRoomNumbers.includes(cardNumber);
+    });
 
-  }
+    toHide.forEach(room => {
+      room.classList.add('hidden')
+    });
+
+  },
+
+  filterRoomByType: (roomType) => {
+    let roomCards = Array.from(document.querySelectorAll('.room-card'));
+
+    roomCards.forEach(card => {
+      const cardType = card.dataset.roomType;
+      if(roomType !== cardType) {
+        card.classList.add('hide')
+      } else {
+        card.classList.remove('hide')
+      }
+    })
+  },
```

Comment:
This function could be improved using jQuery:
```javascript
$(".room-card").hide().filter(`[data-room-type="${cardType}"]`).show();
```

Reply:
Okay implemented the above and it worked! Danke